test/modules/EdgeHubRestartTester/src/Program.cs

```diff
@@ -0,0 +1,161 @@
+
+            Logger.LogInformation($"EdgeHubRestartTester delay start for {Settings.Current.TestStartDelay}.");
+            await Task.Delay(Settings.Current.TestStartDelay);
+
+            (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(Settings.Current.RestartPeriod - TimeSpan.FromMinutes(1), Logger);
+
+            ServiceClient iotHubServiceClient = null;
+            ModuleClient msgModuleClient = null;
+            ModuleClient dmModuleClient = null;
+
+            try
+            {
+                iotHubServiceClient = ServiceClient.CreateFromConnectionString(Settings.Current.ServiceClientConnectionString);
+
+                if (Settings.Current.MessageEnable)
+                {
+                    msgModuleClient = await ModuleUtil.CreateModuleClientAsync(
+                        Settings.Current.TransportType,
+                        ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                        ModuleUtil.DefaultTransientRetryStrategy,
+                        Logger);
+                    msgModuleClient.OperationTimeoutInMilliseconds = Settings.Current.SdkRetryTimeout;
+                }
+
+                if (Settings.Current.DirectMethodEnable)
+                {
+                    dmModuleClient = await ModuleUtil.CreateModuleClientAsync(
+                        Settings.Current.TransportType,
+                        ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                        ModuleUtil.DefaultTransientRetryStrategy,
+                        Logger);
+                }
+
+                DateTime testStart = DateTime.UtcNow;
+                DateTime testExpirationTime = testStart + Settings.Current.TestDuration;
+
+                while ((!cts.IsCancellationRequested) && (DateTime.UtcNow < testExpirationTime))
+                {
+                    (DateTime restartTime, HttpStatusCode restartStatus) = await RestartModules(iotHubServiceClient);
+                    DateTime eachTestExpirationTime = restartTime.Add(Settings.Current.RestartPeriod);
+
+                    // Increment the counter when issue an edgeHub restart
+                    restartCount++;
+
+                    // Setup Message Task
+                    Task sendMessageTask = Task.CompletedTask;
+                    if (Settings.Current.MessageEnable)
+                    {
+                        sendMessageTask =
+                            new MessageEdgeHubConnector(
+                                msgModuleClient,
+                                batchId,
+                                eachTestExpirationTime,
+                                cts.Token,
+                                restartTime,
+                                restartStatus,
+                                restartCount,
+                                Logger)
+                            .ConnectAsync();
```

Comment:
why ConnectAsync?  either sending message or calling DM?   Maybe name it as MessageEdgeHubTester and rename ConnectAsync to StartAsync.

Reply:
`ConnectAsync()` -> `StartAsync()`